ssz-rs/src/list.rs

```diff
@@ -1,26 +1,33 @@
 use crate::de::{deserialize_homogeneous_composite, Deserialize, DeserializeError};
+use crate::lib::*;
 use crate::merkleization::{
     merkleize, mix_in_length, pack, MerkleCache, MerkleizationError, Merkleized, Node,
     BYTES_PER_CHUNK,
 };
 use crate::ser::{serialize_composite, Serialize, SerializeError};
 use crate::{SimpleSerialize, SimpleSerializeError, Sized};
 #[cfg(feature = "serde")]
-use serde::ser::SerializeSeq;
-use std::iter::Enumerate;
+use core::marker::PhantomData;
 #[cfg(feature = "serde")]
-use std::marker::PhantomData;
-use std::ops::{Deref, Index, IndexMut};
-use std::slice::SliceIndex;
-use std::{fmt, slice};
-use thiserror::Error;
-
-#[derive(Error, Debug)]
-pub enum Error {
-    #[error("{provided} elements given that exceeds the list bound of {expected}")]
+use serde::ser::SerializeSeq;
+
+#[derive(Debug)]
+pub enum ListError {
     IncorrectLength { expected: usize, provided: usize },
 }
 
+impl Display for ListError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
```

Comment:
same thing on `core::fmt::Result` -- if we need to do something more verbose, I would actually just leave these all as is!

Reply:
Nice catch, a quick fix.